### PR TITLE
[Rectify] [Php81] Enable Rectify on Intersection Type only

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules/Php70/EregToPcreTransformer.php [hunk 1]
```diff
@@ -281,7 +281,7 @@ private function _ere2pcre_escape(string $content): string
     /**
      * @param mixed[] $r
      */
-    private function processCurlyBracket(string $s, int $i, array &$r, int $rr): int
+    private function processCurlyBracket(string $s, int $i, array & $r, int $rr): int
```
Comment (TomasVotruba) [inline rules/Php70/EregToPcreTransformer.php]: How is this happening?
Should be untouched.

Comment (samsonasik) [reply rules/Php70/EregToPcreTransformer.php]: this is ECS run on php 8.1, not by rector

Comment (samsonasik) [reply rules/Php70/EregToPcreTransformer.php]: the ECS fix diff shows the following on php 8.1 : https://github.com/rectorphp/rector-src/runs/4416719483?check_suite_focus=true#step:7:38 

```diff
     ---------- begin diff ----------
@@ -206,7 +206,7 @@
     /**
      * @param mixed[] $r
      */
-    private function processBracket(string $content, int $i, int $l, array &$r, int $rr): int
+    private function processBracket(string $content, int $i, int $l, array & $r, int $rr): int
     {
         // special case
         if ($i + 1 < $l && $content[$i + 1] === ')') {
@@ -281,7 +281,7 @@
     /**
      * @param mixed[] $r
      */
-    private function processCurlyBracket(string $s, int $i, array &$r, int $rr): int
+    private function processCurlyBracket(string $s, int $i, array & $r, int $rr): int
     {
         $ii = strpos($s, '}', $i);
         if ($ii === false) {
    ----------- end diff -----------


Applied checkers:

 * PhpCsFixer\Fixer\FunctionNotation\FunctionTypehintSpaceFixer
 * PhpCsFixer\Fixer\Operator\ConcatSpaceFixer
 * PhpCsFixer\Fixer\Operator\NotOperatorWithSuccessorSpaceFixer
 * PhpCsFixer\Fixer\Operator\UnaryOperatorSpacesFixer
 *
```

**Comment by samsonasik** [comment]
rebased.

File: rules/Php70/EregToPcreTransformer.php (hunk 1 above)
Comment (TomasVotruba) [reply rules/Php70/EregToPcreTransformer.php]: The responsible rule should be ignored in ecs.php

Comment (samsonasik) [reply rules/Php70/EregToPcreTransformer.php]: I will check in separate PR.

**Comment by samsonasik** [comment]
rebased.

**Comment by samsonasik** [comment]
rebased.

**Review by TomasVotruba — APPROVED** [review]

**Pull request merged**